python/modules/IcePy/Operation.cpp

```diff
@@ -319,27 +300,18 @@ struct OperationObject
     OperationPtr* op;
 };
 
-struct DoneCallbackObject
-{
-    PyObject_HEAD
-    UpcallPtr* upcall;
-    PyObject* coroutine;
-};
-
 struct DispatchCallbackObject
 {
     PyObject_HEAD
     UpcallPtr* upcall;
 };
 
-struct AsyncResultObject
+struct AsyncInvocationContextObject
 {
     PyObject_HEAD
-    Ice::AsyncResultPtr* result;
     InvocationPtr* invocation;
-    PyObject* proxy;
-    PyObject* connection;
-    PyObject* communicator;
+    function<void()>* cancel;
```

Comment:
I don't see where we ever assign a non-null value to invocation.

Reply:
Removed